x/campaign/keeper/msg_server_burn_vouchers.go

```diff
@@ -0,0 +1,48 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/x/campaign/types"
+)
+
+func (k msgServer) BurnVouchers(goCtx context.Context, msg *types.MsgBurnVouchers) (*types.MsgBurnVouchersResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	campaign, found := k.GetCampaign(ctx, msg.CampaignID)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrCampaignNotFound, "%d", msg.CampaignID)
+	}
+
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, spnerrors.Criticalf("can't parse creator address %s", err.Error())
+	}
+
+	// transfer vouchers
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, creator, types.ModuleName, msg.Vouchers); err != nil {
+		return nil, spnerrors.Criticalf("can't send burned coins %s", err.Error())
```

Comment:
This is not a critical error because the message signer balance has not been checked previously and can't be statically verified